File: Java/355.java
import java.util.Scanner;

/**
 *
 * 355 - Gregorio XIII - Iván - https://github.com/wildfireOfMine
 */
public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
       	int prueba = sc.nextInt();
        for (int i = 1; i <= prueba; i++) {
            int	ano = sc.nextInt();
            if ((ano % 4 == 0) && (ano % 100 != 0 || ano % 400 == 0)) {
                System.out.println("29");
            } else {
                System.out.println("28");
            }
        }

    }

}

File: C++/355.cpp
#include <iostream>

using namespace std;

// 355 - Gregorio XIII - Golgotha's Melancholy - https://github.com/Golgotha-sMelancholy

int main()
{
    int loop = 0;
    cin >> loop;
    for (int i = 0; i<loop; i++) {
        int year = 0;
        cin >> year;
        if ((year % 4 == 0) && (year % 100 != 0 || year % 400 == 0)) {
            cout << "29" << endl;
        } else {
            cout << "28" << endl;
        }

    }
}
